luigi/execution_summary.py

```diff
@@ -32,6 +33,66 @@ class execution_summary(luigi.Config):
+    Here are the codes and what they mean:
+
+    =============================  ==========================================================
+    Status Code Name               Meaning
+    =============================  ==========================================================
+    SUCCESS                        There were no failed tasks or missing dependencies
+    SUCCESS_WITH_RETRY             There were failed tasks but they all succeeded in a retry
+    FAILED                         There were failed tasks
+    FAILED_AND_SCHEDULING_FAILED   There were failed tasks and tasks whose scheduling failed
+    SCHEDULING_FAILED              There were tasks whose scheduling failed
+    NOT_RUN                        There were tasks that were not granted run permission by the scheduler
+    MISSING_EXT                    There were missing external dependencies
+    =============================  ==========================================================
+
+    """
+    SUCCESS = (":)", "there were no failed tasks or missing dependencies")
+    SUCCESS_WITH_RETRY = (":)", "there were failed tasks but they all succeeded in a retry")
+    FAILED = (":(", "there were failed tasks")
+    FAILED_AND_SCHEDULING_FAILED = (":(", "there were failed tasks and tasks whose scheduling failed")
+    SCHEDULING_FAILED = (":(", "there were tasks whose scheduling failed")
+    NOT_RUN = (":|", "there were tasks that were not granted run permission by the scheduler")
+    MISSING_EXT = (":|", "there were missing external dependencies")
+
+
+class LuigiRunResult(object):
+    """
+    The result of a call to build/run when passing the detailed_summary=True argument.
+
+    Attributes:
+        - summary_text_one_line (str): One line summary of the progress.
+        - summary_text (str): Detailed summary of the progress.
+        - status (LuigiStatusCode): Luigi Status Code. See :class:`~luigi.execution_summary.LuigiStatusCode` for what these codes mean.
+        - worker (luigi.worker.worker): Worker object. See :class:`~luigi.worker.worker`.
+        - execution_succeeded (bool): Boolean which is *True* if finally, there were no failed tasks.
+        - scheduling_succeeded (bool): Boolean which is *True* if all the tasks were scheduled without errors.
+
+    """
+    def __init__(self, worker, worker_add_run_status=True):
+        self.worker = worker
+        summary_dict = _summary_dict(worker)
+        self.summary_text = _summary_wrap(_summary_format(summary_dict, worker))
+        self.status = _tasks_status(summary_dict)
+        self.summary_text_one_line = _progress_summary(self.status)
+        self.scheduling_succeeded = worker_add_run_status
+
+    # This function makes this class subscriptable (self.worker is used at a few places)
+    def __getitem__(self, key):
+        return getattr(self, key, None)
+
+    def __str__(self):
+        return "STATUS: {0}".format(self.status)
+
+    def __repr__(self):
+        return self.__str__()
```

Comment:
This is not so gonna help debugging. Maybe try to follow the advice in this [excellent Stack Overflow thread](https://stackoverflow.com/a/2626364/621449)?

Reply:
Wow! Nice thread 😮